### PR TITLE
feat: Add support for not specifying a platform to build for

### PR DESCRIPTION
**Pull request by DavidBrunow**
Resolves https://github.com/tuist/tuist/issues/YYY

### Short description 📝

This pull request is meant to aid a conversation about potential ways to implement support for "universal frameworks".

Should we decide to move forward with this effort I will need to add tests.

### Solution 📦

This solution leaves build settings related to the platform empty so that XCConfig files can be used to provide the settings.

### Implementation 👩‍💻👨‍💻

> Detail in a checklist the steps that you took to implement the PR.

- [ ] Step 1
- [ ] Step 2

### CODE REVIEW

File: Sources/TuistGenerator/Linter/GraphLinter.swift [hunk 1]
```diff
@@ -261,6 +261,7 @@ public class GraphLinter: GraphLinting {
             LintableTarget(platform: .iOS, product: .appExtension),
             //            LintableTarget(platform: .iOS, product: .messagesExtension),
             LintableTarget(platform: .iOS, product: .stickerPackExtension),
+            LintableTarget(platform: .notSpecified, product: .framework),
```
Comment (DavidBrunow) [inline Sources/TuistGenerator/Linter/GraphLinter.swift:264]: Probably need to add these entries on more places.

**Comment by DavidBrunow** [comment]
This pull request is meant to aid a conversation about potential ways to implement support for "universal frameworks".

Should we decide to move forward with this effort I will need to add tests.

File: Sources/TuistGenerator/Generator/ConfigGenerator.swift [hunk 2]
```diff
@@ -191,8 +191,14 @@ final class ConfigGenerator: ConfigGenerating {
         if let entitlements = target.entitlements {
             settings["CODE_SIGN_ENTITLEMENTS"] = .string("$(SRCROOT)/\(entitlements.relative(to: sourceRootPath).pathString)")
         }
-        settings["SDKROOT"] = .string(target.platform.xcodeSdkRoot)
```
Comment (DavidBrunow) [inline Sources/TuistGenerator/Generator/ConfigGenerator.swift:194]: This is the meat of the change -- `SDKROOT` and `SUPPORTED_PLATFORMS` are not set which means that values set in XCConfig files can be used.

**Comment by pepibumur** [comment]
Hey @DavidBrunow. Thanks for bringing this up. I understand your motivation behind adding support for specifying certain attributes in your .xconfig files, however, we purposely made them explicit in the manifest because Tuist can easily read the value for providing some convenient features like linting the project, without having to traverse the `.xcconfig` files or generating the project and them using `xcodebuild` and its `-showBuildSettings` argument.

If the need here is supporting universal frameworks, why not build that into Tuist's API and logic, instead of letting users figure out what that means in terms of build settings? From the user point of view, I'd find it way easier to have a declarative API where I can say: "this is a framework for the following platforms X, Y, Z", than having to figure out and ask around how to do that in Tuist. @ollieatkinson had a good idea around how that interface could look so why not exploring that alternative instead. I'm happy to take a stab at it.

**Comment by DavidBrunow** [comment]
Yep definitely agree that it would be better if the XCConfigs are not necessary. I was planning on taking a stab at implementing what you are talking about next unless you’d prefer to do it.

**Comment by pepibumur** [comment]
Hey @DavidBrunow, assuming you'll work on the new approach as part of a new PR, I'll close this one. Feel free to re-open it if that's not the case.

**Pull request closed without merging**